hash.c

```diff
@@ -4160,13 +4160,42 @@ rb_hash_update_block_i(VALUE key, VALUE value, VALUE hash)
  *    update(*other_hashes) -> self
  *    update(*other_hashes) { |key, old_value, new_value| ... } -> self
  *
- *  Like Hash#merge, but modifies and returns +self+ instead of a new hash:
+ *  Updates values and/or adds entries to +self+; returns +self+.
  *
- *    season = {AB: 75, H: 20, HR: 3, SO: 17, W: 11, HBP: 3}
- *    today = {AB: 3, H: 1, W: 1}
- *    yesterday = {AB: 4, H: 2, HR: 1}
- *    season.update(yesterday, today) {|key, old_value, new_value| old_value + new_value }
- *    # => {AB: 82, H: 23, HR: 4, SO: 17, W: 12, HBP: 3}
+ *  Each argument +other_hash+ in +other_hashes+ must be a hash.
+ *
+ *  With no block given, for each successive entry +key+/+new_value+ in each successive +other_hash+:
+ *
+ *  - If +key+ is in +self+, sets <tt>self[key] = new_value</tt>, whose position is unchanged:
+ *
+ *      h0 = {foo: 0, bar: 1, baz: 2}
+ *      h1 = {bat: 3, foo: -1}
+ *      h2 = {bat: 4, bar: -1}
+ *      h0.update(h1, h2) # => {foo: -1, bar: -1, baz: 2, bat: 4}
+ *      h0.update(h2, h1) # => {foo: -1, bar: -1, baz: 2, bat: 3}
```

Comment:
These examples are too complicated IMO. It's also showing what happens when a new entry is added. That's covered by the next section. I think we can simplify it to something like:

```suggestion
 *      h0 = {foo: 0, bar: 1, baz: 2}
 *      h1 = {bar: 3, foo: -1}
 *      h0.update(h1) # => {foo: -1, bar: 3, baz: 2}
```